### ISSUE TITLE
Conan exporting missing file name specifier

### ISSUE THREAD

**Issue by bedeho** [conan-io/conan#1147]
I really need to be able to specify the file name of a given conan recipe file, but currently you can only provide the path.

```
conan export -h
usage: conan export [-h] [--path PATH] [--keep-source] user

Copies the package recipe (conanfile.py and associated files) to your local
cache. From the local cache it can be shared and reused in other projects.
Also, from the local cache, it can be uploaded to any remote with the "upload"
command.

positional arguments:
  user                  user_name[/channel]. By default, channel is "testing",
                        e.g., phil or phil/stable

optional arguments:
  -h, --help            show this help message and exit
  --path PATH, -p PATH  Optional. Folder with a conanfile.py. Default current
                        directory.
  --keep-source, -k     Optional. Do not remove the source folder in the local
                        cache. Use for testing purposes only

```

**Comment by memsharded** [comment]
What would be the use case for this? Do you want to have package recipes like ``conanfile_mypkg1.py``, ``conanfile_mypkg2.py``, living in the same folder? Why not just having them in different folders ``mypkg1/conanfile.py``, ``mypkg2/conanfile.py``, and then doing ``$ conan export --path=mypkg1``?

**Comment by bedeho** [comment]
You are correct that I have a set of conan files together in one folder currently. They are all using some common python code which is living in a hypothetical file `base.py`. Thus I have the following options

- (if `conan export` is updated): They can all live flat together, and each conan file can set `exports=base.py`
- They can be placed in their on new subfolder, and each conan file can set `exports=../base.py`. However, now each conan file has two contradictory constriants on how to import `base.py`. In user space, when being exported, relative importing w.r.t. parent folder must be done. In the cache, the importing will be in the same folder.
- The `base.py` can be turned into its own conan recipe, as you show here
http://conanio.readthedocs.io/en/latest/howtos/python_code_reuse.html. However, the problem here is that we have a very large number of such _sets_ of conanfiles, so we would need to have a large number of such intermediary conan recipes, which we would rather prefer to avoid.

I am barely familiar with python, so I may be missing something, but in general, I think it also makes sense to give the user this freedom, as there can be many other scenarios where this flexibility may be the easiest solution.

**Comment by memsharded** [comment]
Ok, great, makes sense, I just wanted to learn about the usage.
IMO, it can be done, I would suggest using ``--filename`` option because is the one used by ``conan install``.
Shouldn't be difficult to provide, marking it as easy to check if someone can contribute it. Next release is already scheduled, if we are good towards the end of the iteration, I will try to include this feature.

**Comment by bedeho** [comment]
Fantastic, thank you!

**Comment by memsharded** [comment]
Implemented, will be released in 0.22.

**Comment by bedeho** [comment]
Great, Thanks!

**Comment by lasote** [comment]
It has been released in conan 0.22.0. thanks!
